pkg/sql/tablemetadatacache/util/test_utils.go

```diff
@@ -13,7 +13,46 @@ type TestingKnobs struct {
 	OnJobStart func()
 	// onJobComplete is called when the job completes
 	OnJobComplete func()
+	// aostClause overrides the AS OF SYSTEM TIME clause in queries used in
+	// table metadata update job.
+	aostClause string
+	// TableMetadataUpdater overrides the ITableMetadataUpdater used in
+	// tableMetadataUpdateJobResumer.Resume
+	TableMetadataUpdater ITableMetadataUpdater
 }
 
 // ModuleTestingKnobs implements base.ModuleTestingKnobs interface.
 func (*TestingKnobs) ModuleTestingKnobs() {}
+
+// GetAOSTClause returns the appropriate AS OF SYSTEM TIME clause to be
+// used when reading from tables in the job run
+func (knobs *TestingKnobs) GetAOSTClause() string {
+	if knobs != nil {
+		return knobs.aostClause
+	}
+
+	return "AS OF SYSTEM TIME follower_read_timestamp()"
+}
+
+// CreateTestingKnobs creates a testing knob in the unit tests.
+//
+// Note: The table metadata update jopb uses follower read (AS OF SYSTEM TIME
+// follower_read_timestamp()) to ensure that contention between reads and writes
+// is minimized.
+//
+// However, in a new cluster in unit tests, system tables are created using the
+// migration framework. The migration framework goes through a list of
+// registered migrations and creates the stats system tables. By using follower
+// read, we shift the transaction read timestamp far enough to the past. This
+// means it is possible in the unit tests, the read timestamp would be chosen to
+// be before the creation of the stats table. This can cause 'descriptor not
```

Comment:
I think we just mean system table here instead of stats table.

Reply:
Done